src/memory/AmdFlash.hh

```diff
@@ -33,12 +33,19 @@ public:
 		bool oddZero : 1 = false;      // odd bytes are zero, not mirrored
 		size_t readMask = 0x03;        // read address mask
 	};
+	struct Region {
+		size_t count = 0;
+		power_of_two<size_t> size = 1;
+	};
+	struct Geometry {
+		std::array<Region, 4> regions;
```

Comment:
Initially I was wondering what this '4' means. But it's arbitrary, right? So far for (M29W800DB) 4 is the maximum number of regions we need.

Reply:
In the CFI query format there are up to four region entries, so this maximum is assured.

Reply:
I’m changing it to `static_vector<Region, 4> regions;`, this will improve the logic in `peekCFI()` and should get rid of GCC’s “either all initializer clauses should be designated or none of them should be” complaints.

Reply:
And same for `AutoSelect::device`, changing it to `static_vector<uint8_t, 2> device;`.